lib/league_stats.rb

```diff
@@ -1,71 +1,238 @@
 require 'CSV'
-
-require_relative './games'
-require_relative './teams'
 require_relative './game_teams'
+require_relative './game'
+require_relative './team'
 
 class LeagueStats
 
+  attr_reader :game_teams,
+              :teams,
+              :games
+
+  def initialize(filepath1, filepath2, filepath3)
+    @game_teams = []
+    @games      = []
+    @teams      = []
+    load_game_teams(filepath1)
+    load_games(filepath2)
+    load_teams(filepath3)
+  end
 
-  def initialize(filepath)
-    #yeah, rename the below
-    @game_teams_stuff = []
-    load_game_teams(filepath)
+
+  def load_game_teams(filepath1)
+    CSV.foreach(filepath1, headers: true, header_converters: :symbol) do |data|
+      @game_teams << GameTeams.new(data)
+    end
   end
 
-  def all
-    @game_teams_stuff
+  def load_games(filepath2)
+    CSV.foreach(filepath2, headers: true, header_converters: :symbol) do |data|
+      @games << Game.new(data)
+    end
+  end
+
+  def load_teams(filepath3)
+    CSV.foreach(filepath3, headers: true, header_converters: :symbol) do |data|
+      @teams << Team.new(data)
+    end
   end
 
-  def load_game_teams(filepath)
-    CSV.foreach(filepath, headers: true, header_converters: :symbol) do |data|
-      @game_teams_stuff << GameTeams.new(data)
+
+  def game_teams_find_by_game_id(game_id)
+    @game_teams.find do |season_stat|
+      season_stat.game_id == game_id
     end
   end
 
-  def find_by_game_id(game_id)
-    @game_teams_stuff.find do |season_stat|
-      # require 'pry'; binding.pry
+  def games_find_by_game_id(game_id)
+    @games.find do |season_stat|
       season_stat.game_id == game_id
     end
   end
-  # #LEAGUE STATS
+
+  def teams_find_by_team_id(team_id)
+    @teams.find do |season_stat|
+      season_stat.team_id == team_id
+    end
+  end
+
+  def largest_hash_key(hash)
+    hash.max_by{|k,v| v}
+  end
+
+  def smallest_hash_key(hash)
+    hash.min_by{|k,v| v}
+  end
+
+
+  # #LEAGUE STATS METHODS
   #
   # # 	Total number of teams in the data.  INTEGER
-  #   def count_of_teams
-  #
-  #   end
+    def count_of_teams
+      @teams.count
+    end
   #
   # #  Name of the team with the highest average number of goals scored per game across all seasons.  STRING
-  #   def best_offense
-  #
-  #   end
+    def best_offense
```

Comment:
I think this could be a module !!! As I can see this being used in many other class stats.